src/BaGet/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -118,42 +128,46 @@ public static IServiceCollection ConfigureHttpServices(this IServiceCollection s
             return services;
         }
 
-        public static IServiceCollection ConfigureStorageProviders(
-            this IServiceCollection services,
-            IConfiguration configuration)
+        public static IServiceCollection ConfigureStorageProviders(this IServiceCollection services)
         {
+            services.AddFileStorageService();
+            services.AddBlobStorageService();
+
             services.AddTransient<IPackageStorageService, PackageStorageService>();
 
-            var storageOptions = configuration.Get<BaGetOptions>().Storage;
-            switch (storageOptions.Type)
+            services.AddTransient<IStorageService>(provider =>

```

Comment:
@LordMike this goes back to registering all storage implementations to the dependency injection container, and then using `IStorageService` to decide which implementation to use. This ensures that options aren't used until after they've been validated.